### DISCUSSION TITLE
Instrumentation of Kafka Clients : Add custom Attributes Extractor

### DISCUSSION THREAD

**Discussion by adityasc94** [open-telemetry/opentelemetry-java-instrumentation#9516] [Q&A]
I am using the interceptors provided by Kafka clients instrumentation library in a simple Kafka Producer by following the example in: https://github.com/ppatierno/kafka-opentelemetry/tree/main/kafka-producer

I want send custom attributes to the span.

The KafkaTelemetryBuilder class has a method "addProducerAttributesExtractors" to add extra AttributesExtractor:
[opentelemetry-java-instrumentation/instrumentation/kafka/kafka-clients/kafka-clients-2.6/library/src/main/java/io/opentelemetry/instrumentation/kafkaclients/v2_6/KafkaTelemetryBuilder.java](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/e8464f696082b2181e04d57cfc48a1e79ea70ac2/instrumentation/kafka/kafka-clients/kafka-clients-2.6/library/src/main/java/io/opentelemetry/instrumentation/kafkaclients/v2_6/KafkaTelemetryBuilder.java#L38-L43)

I would like to know if there is a way to use this approach along with the interceptors.

**Comment by laurit** [comment]
You can copy the interceptor class and modify the way it builds the `KafkaTelemetry` instance.

**Reply by adityasc94** [reply]
Hi @laurit, thank you for the feedback. I tried using this approach, but I am not able to call the "buildAndInjectSpan" method of KafkaTelemetry instance from my custom interceptor as the method is having "default" scope which restricts access outside the package.

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/e8464f696082b2181e04d57cfc48a1e79ea70ac2/instrumentation/kafka/kafka-clients/kafka-clients-2.6/library/src/main/java/io/opentelemetry/instrumentation/kafkaclients/v2_6/KafkaTelemetry.java#L177
